src/mono/wasm/runtime/binding_support.js

```diff
@@ -117,10 +124,26 @@ var BindingSupportLib = {
 				return BINDING.bind_method (method, 0, signature, "BINDINGS_" + method_name);
 			};
 
+			this._are_promises_supported = ((typeof Promise === "object") || (typeof Promise === "function")) && (typeof Promise.resolve === "function");
+
+			this._empty_string = "";
+			this._empty_string_ptr = 0;
+			this._interned_string_full_root_buffers = [];
+			this._interned_string_current_root_buffer = null;
+			this._interned_string_current_root_buffer_count = 0;
+			this._interned_string_table = new Map ();
+			this._managed_pointer_to_interned_string_table = new Map ();
+			this._custom_marshaler_info_cache = new Map ();
+
+			// HACK: This method needs to be the absolute first one we bind, because
+			//  the process of binding other methods relies on it.
+			this.make_marshal_signature_info = bind_runtime_method ("MakeMarshalSignatureInfo", "iiii");
```

Comment:
Can we add assert/check instead of comment?

Reply:
Yes, I think I can figure out a way to do that. Great idea